Copy get attr nodes to submodules during partition fusion

madhu-fb opened the pull request:
Summary: Prior to this change, partition attributes were copied as input(placeholder) nodes into the subgraph. This is not ideal for lowering - copy them as get_attr nodes instead.

Test Plan: test_fx_passes.py

Differential Revision: D40950356

angelayi commented on torch/fx/passes/utils/fuser_utils.py:
```diff
@@ -108,14 +108,16 @@ def fuse_as_graphmodule(gm: GraphModule,
+            # Copy get_attr nodes as get_attr nodes. lift_graph_as_submodule call below ensures that
+            # the data is copied efficiently. The get_attr node is left as-is in the original graph
+            # in case it is used elsewhere outside the submodule.
+            get_attr_node = subgraph.get_attr(x.name, type_expr=x.type)
```
I think using the subgraph.copy function might be better since it'll also copy over any metadata attached to that node. Thoughts? @SherlockNoMad 
```
get_attr_node = subgraph.node_copy(x)
```

SherlockNoMad replied:
Yes, node_copy is better here, as it will also copy the node.meta field.

madhu-fb replied:
Thanks, I'll update with node_copy()

SherlockNoMad commented on test/test_fx_passes.py:
```diff
@@ -310,12 +333,52 @@ def test_fuser_util(self, partition):
+    @parametrize("partition", [
+        [['conv1', 'conv2', 'add_2', 'conv3']],
+    ])
+    def test_fuser_util_with_get_attr(self, partition):
```
nitpick, this is kinda hard to read what it's trying to test... 
Do you really need to construct a quantized model for testing the get_attr copying behavior?

madhu-fb committed:
Copy get attr nodes to submodules during partition fusion (#88608)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/88608

Prior to this change, partition attributes were copied as input(placeholder) nodes into the subgraph. This is not ideal for lowering - copy them as get_attr nodes instead.

Test Plan: test_fx_passes.py

Differential Revision: D40950356

fbshipit-source-id: 4e8f88e13bd176614610c7bb16c2ab886fcb9ba0

angelayi commented:
Looks good to me -- will wait for Sherlock's approval
